In [1]:
from ultralytics import YOLO


model = YOLO(f"yolo11n.yaml")
model = YOLO(f"yolo11n.pt")
results = model.train(data="trash.yaml", epochs=3,imgsz=1024)
results = model.val()
  

Ultralytics 8.3.15 🚀 Python-3.10.13 torch-1.12.1+cu116 CUDA:0 (Tesla V100-SXM2-32GB, 32501MiB)
WARNING ⚠️ Upgrade to torch>=2.0.0 for deterministic training.
engine/trainer: task=detect, mode=train, model=yolo11n.pt, data=trash.yaml, epochs=3, time=None, patience=100, batch=16, imgsz=1024, save=True, save_period=-1, cache=False, device=None, workers=8, project=None, name=train10, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False,

wandb: Using wandb-core as the SDK backend. Please refer to https://wandb.me/wandb-core for more information.
wandb: W&B API key is configured. Use `wandb login --relogin` to force relogin


Freezing layer 'model.23.dfl.conv.weight'
AMP: running Automatic Mixed Precision (AMP) checks with YOLO11n...
AMP: checks passed ✅


train: Scanning /data/ephemeral/home/level2-objectdetection-cv-20/dataset/yolo_train80/labels.cache... 4337 images, 0 backgrounds, 0 corrupt: 100%|██████████| 4337/4337 [00:00<?, ?it/s]

train: WARNING ⚠️ /data/ephemeral/home/level2-objectdetection-cv-20/dataset/yolo_train80/images/0001.jpg: 8 duplicate labels removed
train: WARNING ⚠️ /data/ephemeral/home/level2-objectdetection-cv-20/dataset/yolo_train80/images/0003.jpg: 2 duplicate labels removed
train: WARNING ⚠️ /data/ephemeral/home/level2-objectdetection-cv-20/dataset/yolo_train80/images/0004.jpg: 6 duplicate labels removed
train: WARNING ⚠️ /data/ephemeral/home/level2-objectdetection-cv-20/dataset/yolo_train80/images/0005.jpg: 22 duplicate labels removed
train: WARNING ⚠️ /data/ephemeral/home/level2-objectdetection-cv-20/dataset/yolo_train80/images/0008.jpg: 7 duplicate labels removed
train: WARNING ⚠️ /data/ephemeral/home/level2-objectdetection-cv-20/dataset/yolo_train80/images/0013.jpg: 48 duplicate labels removed
train: WARNING ⚠️ /data/ephemeral/home/level2-objectdetection-cv-20/dataset/yolo_train80/images/0018.jpg: 22 duplicate labels removed
train: WARNING ⚠️ /data/ephemeral/home/level2-objectdetection-cv-2

albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01), CLAHE(p=0.01, clip_limit=(1, 4.0), tile_grid_size=(8, 8))


val: Scanning /data/ephemeral/home/level2-objectdetection-cv-20/dataset/yolo_val20/labels.cache... 1598 images, 0 backgrounds, 0 corrupt: 100%|██████████| 1598/1598 [00:00<?, ?it/s]

val: WARNING ⚠️ /data/ephemeral/home/level2-objectdetection-cv-20/dataset/yolo_val20/images/0001.jpg: 16 duplicate labels removed
val: WARNING ⚠️ /data/ephemeral/home/level2-objectdetection-cv-20/dataset/yolo_val20/images/0005.jpg: 22 duplicate labels removed
val: WARNING ⚠️ /data/ephemeral/home/level2-objectdetection-cv-20/dataset/yolo_val20/images/0014.jpg: 5 duplicate labels removed
val: WARNING ⚠️ /data/ephemeral/home/level2-objectdetection-cv-20/dataset/yolo_val20/images/0045.jpg: 19 duplicate labels removed
val: WARNING ⚠️ /data/ephemeral/home/level2-objectdetection-cv-20/dataset/yolo_val20/images/0051.jpg: 13 duplicate labels removed
val: WARNING ⚠️ /data/ephemeral/home/level2-objectdetection-cv-20/dataset/yolo_val20/images/0054.jpg: 8 duplicate labels removed
val: WARNING ⚠️ /data/ephemeral/home/level2-objectdetection-cv-20/dataset/yolo_val20/images/0062.jpg: 28 duplicate labels removed
val: WARNING ⚠️ /data/ephemeral/home/level2-objectdetection-cv-20/dataset/yolo_val20/images/

Plotting labels to runs/detect/train10/labels.jpg... 
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: AdamW(lr=0.000714, momentum=0.9) with parameter groups 81 weight(decay=0.0), 88 weight(decay=0.0005), 87 bias(decay=0.0)
TensorBoard: model graph visualization added ✅
Image sizes 1024 train, 1024 val
Using 8 dataloader workers
Logging results to runs/detect/train10
Starting training for 3 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


        1/3      7.28G      1.189       3.37      1.546          2       1024: 100%|██████████| 272/272 [00:51<00:00,  5.27it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 50/50 [00:18<00:00,  2.70it/s]


                   all       1598      10829      0.304      0.211      0.136     0.0882

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


        2/3      7.65G      1.155      2.714      1.516          4       1024: 100%|██████████| 272/272 [00:46<00:00,  5.89it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 50/50 [00:13<00:00,  3.74it/s]


                   all       1598      10829      0.375      0.243      0.209      0.138

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


        3/3      8.09G      1.099       2.46      1.467          2       1024: 100%|██████████| 272/272 [00:44<00:00,  6.14it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 50/50 [00:16<00:00,  2.97it/s]


                   all       1598      10829      0.425      0.307      0.287      0.197

3 epochs completed in 0.056 hours.
Optimizer stripped from runs/detect/train10/weights/last.pt, 5.6MB
Optimizer stripped from runs/detect/train10/weights/best.pt, 5.6MB

Validating runs/detect/train10/weights/best.pt...
Ultralytics 8.3.15 🚀 Python-3.10.13 torch-1.12.1+cu116 CUDA:0 (Tesla V100-SXM2-32GB, 32501MiB)
YOLO11n summary (fused): 238 layers, 2,584,102 parameters, 0 gradients, 6.3 GFLOPs


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 50/50 [00:18<00:00,  2.63it/s]


                   all       1598      10829      0.426      0.307      0.288      0.198
         General trash        885       1887      0.302      0.164      0.152      0.108
                 Paper        754       2880      0.321      0.517      0.391      0.237
            Paper pack        273        405      0.254      0.168      0.149      0.112
                 Metal        237        362      0.442      0.227       0.23      0.191
                 Glass        132        404      0.388      0.448      0.339      0.204
               Plastic        623       1581      0.287      0.269      0.203       0.13
             Styrofoam        252        612      0.388       0.44      0.383      0.256
               Plastic        792       2455      0.575      0.567      0.596      0.437
               Battery         16         68          1          0      0.227      0.186
              Clothing         81        175      0.303      0.274       0.21      0.119
Speed: 0.8ms preproce

Ultralytics 8.3.15 🚀 Python-3.10.13 torch-1.12.1+cu116 CUDA:0 (Tesla V100-SXM2-32GB, 32501MiB)
YOLO11n summary (fused): 238 layers, 2,584,102 parameters, 0 gradients, 6.3 GFLOPs


val: Scanning /data/ephemeral/home/level2-objectdetection-cv-20/dataset/yolo_val20/labels.cache... 1598 images, 0 backgrounds, 0 corrupt: 100%|██████████| 1598/1598 [00:00<?, ?it/s]

val: WARNING ⚠️ /data/ephemeral/home/level2-objectdetection-cv-20/dataset/yolo_val20/images/0001.jpg: 16 duplicate labels removed
val: WARNING ⚠️ /data/ephemeral/home/level2-objectdetection-cv-20/dataset/yolo_val20/images/0005.jpg: 22 duplicate labels removed
val: WARNING ⚠️ /data/ephemeral/home/level2-objectdetection-cv-20/dataset/yolo_val20/images/0014.jpg: 5 duplicate labels removed
val: WARNING ⚠️ /data/ephemeral/home/level2-objectdetection-cv-20/dataset/yolo_val20/images/0045.jpg: 19 duplicate labels removed
val: WARNING ⚠️ /data/ephemeral/home/level2-objectdetection-cv-20/dataset/yolo_val20/images/0051.jpg: 13 duplicate labels removed
val: WARNING ⚠️ /data/ephemeral/home/level2-objectdetection-cv-20/dataset/yolo_val20/images/0054.jpg: 8 duplicate labels removed
val: WARNING ⚠️ /data/ephemeral/home/level2-objectdetection-cv-20/dataset/yolo_val20/images/0062.jpg: 28 duplicate labels removed
val: WARNING ⚠️ /data/ephemeral/home/level2-objectdetection-cv-20/dataset/yolo_val20/images/


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 100/100 [00:26<00:00,  3.77it/s]


                   all       1598      10829      0.426      0.307      0.288      0.198
         General trash        885       1887      0.303      0.165      0.152      0.107
                 Paper        754       2880       0.32      0.516      0.391      0.236
            Paper pack        273        405      0.257      0.169      0.148      0.111
                 Metal        237        362      0.443      0.226       0.23      0.192
                 Glass        132        404      0.387      0.448      0.339      0.204
               Plastic        623       1581      0.286      0.268      0.203       0.13
             Styrofoam        252        612      0.385      0.438      0.382      0.256
               Plastic        792       2455      0.575      0.568      0.596      0.437
               Battery         16         68          1          0      0.224      0.184
              Clothing         81        175      0.305      0.274       0.21      0.119
Speed: 0.7ms preproce

engine/trainer: 
task=detect, mode=train, model=yolo11m.pt, data=coco8.yaml, epochs=3, time=None, patience=100, batch=16, imgsz=640, save=True, save_period=-1, cache=False, device=None, workers=8, project=None, name=train, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=True, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, bgr=0.0, mosaic=1.0, mixup=0.0, copy_paste=0.0, copy_paste_mode=flip, auto_augment=randaugment, erasing=0.4, crop_fraction=1.0, cfg=None, tracker=botsort.yaml, save_dir=runs/detect/train
